### DISCUSSION TITLE
Setting flow temp on a Worcester Bosch 5000i - command ignored

### DISCUSSION THREAD

**Discussion by bittylicious** [emsesp/EMS-ESP32#2293] [Q&A] [answered]
I'm trying to configure home automation to essentially "call for heat" using EMS commands.

When trying to set the flow rate, e.g. using **call boiler selflowtemp**, it looks successful but the boiler just ignores the command. For instance, if it's set to 75° and I set it to 0 or 10, it will remain set at 75°.

I had a [Bosch Easycontrol ](https://www.worcester-bosch.co.uk/professional/support/literature/bosch-easycontrol-control-key-1) key installed and this is essentially the thermostat, i.e. a connected device tells the boiler what the selflowtemp should be (I can see it going between 0° and 75° when controlled by this). However, even with the key uninstalled, the boiler disregards the selflowtemp command.

Having done some debugging, I see these commands coming through when the key wants to call for heat (which sets the flow temp to 75°C), and when it doesn't need heat (which sets the flow temp to 0°C):

```
000+12:37:54.211 T 144: [emsesp] thermostat(0x18) -> boiler(0x08), W, ?(0x02E0), data: 01 4B 64 00 01
000+12:36:54.792 T 37: [emsesp] thermostat(0x18) -> boiler(0x08), W, ?(0x02E0), data: 01 00 00 00 01
```

These seem to correspond to (but please do correct me!):

```
call system send "0B 08 FF 00 01 E0 01 4B 64 00 01"
call system send "0B 08 FF 00 01 E0 01 00 00 00 01"
```

Has anyone ever seen this before? I'm only just starting my debugging, but it looks like my command can indeed set the flow temp where no other method I've found works.

The key also sends a few other commands which I haven't figured out yet, so would welcome advice:

```
000+12:39:36.503 T 272: [emsesp] thermostat(0x18) -> boiler(0x08), W, ?(0xE7), data: 00 02 00 64
000+12:39:36.565 T 274: [emsesp] thermostat(0x18) -> boiler(0x08), W, ?(0x02EA), data: 00
```

Finally, how would you recommend I actually control this using Home Assistant? Should I be creating some sort of sensor in the EMS-ESP web interface?

Many thanks for your help.

**Comment by proddy** [comment]
Perhaps the EasyControl is overriding the settings? Does changing the `selflowtemp` work with the thermostat disabled? Also note the boiler will only accept a flow temperature value that is lower than the current heating temperature (EMS/Bosch design).

**Comment by bittylicious** [comment]
Hi proddy. Unfortunately even with the EasyControl key physically out, I still cannot change **selflowtemp**. I don't think it's being set back - reading the values immediately after setting them results suggests they didn't change at all.

Noted about the flow temperature having to be lower than the current heating temperature. When I set a boiler temperature, the flow temperature changes to that same temperature on the next EMS update. The problem is that I cannot set the boiler temperature to be below 30°C (I'm aiming for "full off", i.e. zero).

I did read a post-manufacture document a while ago suggesting that if you used an external thermostat (e.g. the EasyControl key), the boiler needs a factory reset when changing this. I don't really want to do this as I'm not confident to recommission the boiler.

**Reply by proddy** [reply]
does the boiler temp change (and max pump %) when you enable the forceheatingoff function in EMS-ESP?

**Reply by bittylicious** [reply]
No, neither changes. The functionality does work though - I it stops the boiler from firing - but it doesn't change any flow temps / max pump percentages. It is a solution, but hacky and I prefer if I can actually force a change of the flow temp which my command seems to do.

**Reply by MichaelDvP** [reply]
@proddy We always set `selflowtemp` with 0x1A telegram. Seems some of the newer ems+ boilers uses 0x2E0 for commands. But without product-id we can not change. Also unclear is offset 2, could be pump power or max burner power.

**Reply by bittylicious** [reply]
When you say you need a product-id, is that something I can provide?

**Comment by bittylicious** [comment, marked as answer]
This does seem to work quite well to simply set the flow temperature. It'll set it, on my system, up to 75°C, which is what I have the max temperature set to.

![Screenshot from 2024-12-15 16-59-13](https://github.com/user-attachments/assets/0c02b60c-d132-4266-93d1-892ec79c1bf4)

Because it's a non-raw value, it can also be nicely set in HomeAssistant.

**Comment by bittylicious** [comment]
Just wanted to quickly add to this.

Unless we regularly send a temperature command to the boiler, it eventually defaults back to 75°C (FF technically, but that's 75°C here). I have had to create an automation in Home Assistant to regularly set the value (FakeEasy_FlowTemp in the above screenshot) to zero when we don't want to call for heat.

Is this how things worked with the old 0x1A telegram? Will there be support for the 0x2E0 telegram to be triggered if we set a required flow temperature in a later version of EMS-ESP?

**Reply by MichaelDvP** [reply]
For the 0x1A the fallback is is after ~2 min to the value of heatingtemp. The ems-esp function `forceheatingoff` sets every minute zero to 0x1A. I think this will be similar for the 0x2E0 telegram.

**Comment by Sbried** [comment]
Then thing with the 0x02E0 telegram is that the whole telegram has to be submitted, not only parts of it. In offset 0 is the request for heat (1=yes, 0=no). Offset 1 is the requested supply temperature as UINT8. Offset 2 is the requested power: I think always sending 100% is ok, then the power will be adjusted to the requested supply temperature. Offset 3 is kind of efficiency requester: 0=heat up in an efficient way, 1=heat up as fast as possible. Finally there is offset 4 which has to do something with season optimizing and can be 1 or 2.
You may try to send always the whole telegram with i.e. 01 [desired temp] 64 00 01 and see what happens

**Comment by bittylicious** [comment]
Thanks for the additional information @MichaelDvP and @Sbried 

How are things supposed to work with EMS-ESP? I assumed you just set the flow temperature, and EMS-ESP handles all the magic (e.g. periodically sending messages to the boiler so it doesn't fall back). If it doesn't do this, how did it work with old 0x1A telegrams?

My feeling is that it should act functionally the same with the new 0x2E0 telegrams, even if we have to read the current state of 0x2E0 before changing the bits we care about.

Finally, what happens when I call by FakeEasy_FlowTemp call above, which is just a UNIT8 with offset 1? Based on what @Sbried says, we can't just update a part of the telegram, so behind the scenes, does it expressly read the value first or rely on its latest update?

**Reply by MichaelDvP** [reply]
No, ems-esp is mainly just converting ems-bus entities to mqtt entities. 
There are only very few special functions that sadly often leads to misunderstandings by users. Periodical sending is only done for `forceheatingoff` and the remote thermostat emulations. 
The 0x2E0 telegram is new to us, i don't know how it works. For 0x1A we do not send the whole telegram, setting flowseltemp just sends 0x1A offset 0. Seems to work (i have a thermostat and never used it). Only forceheatingoff sends the complete telegram with zeros to alos stop the burner.

Just try ot what is working. You can use `system/send` command to send raw telegrams, or create a custom entitiy with raw values to send only the hex data string.

**Reply by MichaelDvP** [reply]
I have reworked the scheduler functions. Now it's possible to send the whole telegram every minute with schedule action.
Use https://github.com/MichaelDvP/EMS-ESP32/releases/tag/latest
- create a custom entity `flowtemp`in RAM
- create a custom entitiy `set_2e0` as EMS-value: device 0x08, telegram 0x2E0, offset 0,  RAW, length 5
- create a timer schedul like this:
![grafik](https://github.com/user-attachments/assets/c0c5e3a0-0d90-4577-8276-e3ba7e6c7005)

Then you can write a value to `custom/flowtemp` and the telegram to 2E0 is send every minute with this value in offset 1.
